modules/javafx.controls/src/main/java/javafx/scene/control/ContextMenu.java

```diff
@@ -244,55 +246,19 @@ public void show(Node anchor, Side side, double dx, double dy) {
         if (getItems().size() == 0) return;
 
         getScene().setNodeOrientation(anchor.getEffectiveNodeOrientation());
-        setAnchorLocation(getAnchorLocation(side, anchor.getEffectiveNodeOrientation()));
-        Bounds anchorBounds = anchor.localToScreen(anchor.getLayoutBounds());
-        double x = getXBySide(anchorBounds, side, anchor.getEffectiveNodeOrientation()) + dx;
-        double y = getYBySide(anchorBounds, side) + dy;
-        doShow(anchor, x, y);
-    }
+        if (getScene().getStylesheets().isEmpty()) {
+            getScene().getStylesheets().setAll(anchor.getScene().getStylesheets());
```

Comment:
I need to verify this, but I presume that the `Scene` of the `ContextMenu` is something that is created by JavaFX (as opposed to something that can be set by the application)? If so, then this might be an OK fix, but we will need to ensure that there are no side effects of doing this.